hooks/pre/31-upgrade_storage_engine.rb

```diff
@@ -0,0 +1,56 @@
+MONGO_ENGINE_COMPLETE = '/etc/foreman-installer/.mongo_engine_upgrade'.freeze
+MONGO_DIR = "/var/lib/mongodb"
+
+# check diskspace
+def disk_space
+  logger.info 'Checking available diskspace in /tmp for upgrade'
+  total_space = `df -k --total --exclude-type=tmpfs`.split("\n")[-1].split[3].to_i
+  mongo_size = File.directory?(@MONGO_DIR) ? `du -s  /var/lib/mongodb/`.split[0].to_i : 0
+  if total_space < mongo_size
+    logger.error 'The free space is less than the size of the Mongo db directory, please add additional space to /tmp and try again, exiting.'
+    kafo.class.exit(1)
+  else
+    logger.info 'Diskspace looks good, continuing to upgrade'
+    migration
+  end
+end
+
+def migration
```

Comment:
Some empty lines throughout this method would help a lot with readability. Breaking segments into logical chunks.